### PR TITLE
Remove telegraf from linux 

### PR DESCRIPTION
**Pull request by Sohamdg081992**
This version uses cpu plugin to collect system level cpu information from container as opposed to exec plugin which collects cpu from otelcollector and me processes indivudually. Exec plugin is not supported in distroless images.

Apart from the below testing, I have also tested the telemetry flow by restarting the otelcol and me process multiple times and it works fine.

------------------------------------
Screenshots for Memory reduction after removing telegraf

Without Telegraf
GrafanaWOTelMem%
GrafanaWOTelRS
GrafanaWOTelDS

With Telegraf
GrafanaWTelMem%
GrafanaWTelRS
GrafanaWTelDS

---------------------------------

Value comparison of Telegraf vs. Fluent Bit( both components were running on the same pods for this testing). We got close values for both.

I deployed the reference app and enabled scraping. Then I increased the number of replicas to 10 for a few minutes and then reduced it to 1 again. The HI RS and HI DS are metric values recorded when this spike was made and Low RS and Low DS are values recorded in other times.

portforward

---------

 Fluent bit

|                             | HI DS        | HI RS        | Low DS       | Low RS       |
|-----------------------------|--------------|--------------|--------------|--------------|
| otelcolVMRSSAvg            | 117068000    | 120916000    | 111536000    | 119460000    |
| otelcolVMRSS95             | 117068000    | 122388000    | 111536000    | 121312000    |
| meVMRSSAvg                 | 44834816     | 41312256     | 58975872     | 48975872     |
| meVMRSS95                  | 45671628     | 48975872     | 48976472     | 48975872     |
| otelcpuUsageAvg(%)         | 52           | 49           | 28           | 48           |
| otelcpuUsage95(%)          | 57           | 53.5         | 30           | 55           |
| meCpuUsageAvg(%)           | 43           | 38.5         | 49.5         | 41           |
| meCpuUsage95(%)            | 58.5         | 79           | 57           | 51.5         |

-----------------------------------

---------

 Telegraf

|                             | HI DS              | HI RS              | Low DS             | Low RS             |
|-----------------------------|--------------------|--------------------|--------------------|--------------------|
| otelcollector_memory_rss_050| 126353408          | 120803328          | 124489728          | 144173056          |
| otelcollector_memory_rss_095| 126353408          | 135784038          | 128032768          | 146022400          |
| metricsextension_memory_rss_050| 44834816       | 41312256           | 36065280           | 55238656           |
| metricsextension_memory_rss_095| 45671628       | 48975872           | 37960908           | 55238656           |
| otelcollector_cpu_usage_050 | 0.321608126437242 | 0.143547678977113  | 0.171110652970617  | 0.190275998078321  |
| otelcollector_cpu_usage_095 | 0.458843069727404 | 0.959536547881522  | 0.286772234370546  | 0.696191806862753  |
| metricsextension_cpu_usage_050 | 0.274309766455724 | 0.401607109869771 | 0.234063760776055 | 0.40123204052676 |
| metricsextension_cpu_usage_095 | 0.485236937139407 | 0.617179716178943 | 0.332776288758776 | 0.609020321347796 |

-----------------------------------

Below are screenshots of all metric types. 

envVars

mecpu
mecpuDimension
meVmrss95
otelcolvmrss95
otelcpu

----------------------------------------

The below section has screenshots of operator related telemetry from both Telegraf and Fluent bit , showing that the dimensions are identical.

Telegraf based screenshots
TelegrafOperatorModel1
TelegrafOperatorModel2

OperatorMetricType

Fluent Bit based screenshots

FBOperatorModel1
FBOperatorModel11
FBOperatorModel22
FBOperatorModel21

--------------------------------------------------------------------------------------

Below are the metric name based screenshots of Telegraf and Fluent Bit, they show both are same.

Telegraf based screenshots:

Target allocator metric names

TelegrafTAMetricName

Prometheus metric names

TelegrafPromMetricName

CPU usage metric names for a process

TelegrafCPUMetricName

Memory usage metric names for a process     

TelegrafMemoryMetricName
 

Fluent Bit based screenshots:

Target allocator metric names
FBOperatorMetricName1
FBOperatorMetricName2
FBOperatorMetricName3

Prometheus metric names

FBPromMetricName

CPU usage metric names for a process
FBCPUMetricName

Memory usage metric names for a process 

FBMemoryMetricName1

### CODE REVIEW

File: otelcollector/fluent-bit/src/telemetry.go [hunk 1]
```diff
@@ -835,3 +841,214 @@ func RecordExportingFailed(records []map[interface{}]interface{}) int {
...
+	return output.FLB_OK
+}
+
+func PushOtelCpuToAppInsightsMetrics(records []map[interface{}]interface{}) int {
+	var totalCpuUsage float64
+	var count int
+
+	duration := time.Duration(meOtelCpuMemoryUsageIntervalSeconds) * time.Second
+	deadline := time.Now().Add(duration)
+
```
Comment (Sohamdg081992) [inline otelcollector/fluent-bit/src/telemetry.go:885]: check for 95 pct?

File: otelcollector/fluent-bit/fluent-bit-parsers.conf [hunk 2]
```diff
@@ -34,3 +34,27 @@
     Time_Key time
     Time_Format %Y-%m-%dT%H:%M:%S
     Time_Keep On
+
+[PARSER]
+    Name proc-parser
+    Format regex
+    Regex ^[^\[]*\[\s*(?<time>[^\s,]+),\s*(?<message>.*)\]$
```
Comment (Sohamdg081992) [inline otelcollector/fluent-bit/fluent-bit-parsers.conf:41]: do spike test with telegraf

File: otelcollector/telegraf/telegraf-prometheus-collector-ds.conf [hunk 3]
```diff
@@ -1,192 +0,0 @@
...
-   exe = "otelcollector"
-   interval = "10s"
-   pid_finder = "pgrep"
-   pid_tag = true
-   name_override = "otelcollector"
-   fieldpass = ["cpu_usage", "memory_rss"]
-   [inputs.procstat.tags]
-#    Computer = "$NODE_NAME"
-#    NodeIp = "$NODE_IP"
-    cpulimit = "$CONTAINER_CPU_LIMIT"
```
Comment (Sohamdg081992) [inline otelcollector/telegraf/telegraf-prometheus-collector-ds.conf:171]: add these customdimensions.

Comment (Sohamdg081992) [reply otelcollector/telegraf/telegraf-prometheus-collector-ds.conf:171]: I have added this. I have attached the screenshots above.

File: otelcollector/fluent-bit/fluent-bit-parsers.conf (hunk 2 above)
Comment (Sohamdg081992) [reply otelcollector/fluent-bit/fluent-bit-parsers.conf:41]: Completed and attached results.

**Review by gracewehner — APPROVED** [review]
Approving but would like someone else to also take a look before this is merged

**Review by bragi92 — APPROVED** [review]

File: otelcollector/fluent-bit/fluent-bit-daemonset.conf [hunk 4]
```diff
@@ -130,6 +130,42 @@
...
+[INPUT]
+    Name cpu
+    Tag  cpu.otelcollector
+    PID  ${OTEL_PID}
+
+[INPUT]
+    Name cpu
+    Tag  cpu.me
+    PID  ${ME_PID}
+
```
Comment (rashmichandrashekar) [inline otelcollector/fluent-bit/fluent-bit-daemonset.conf:168]: this seems to be missing the input plugin for targetallocator - 
[[inputs.prometheus]]
  interval = "5m"
  urls = ["http://ama-metrics-operator-targets.kube-system.svc.cluster.local/metrics"]
  fieldpass = ["opentelemetry_allocator_targets","opentelemetry_allocator_collectors_discovered"]
  metric_version = 2
  url_tag = "scrapeUrl"
  response_timeout = "15s"
  name_override = "target_allocator"

Could you pls add that?

Comment (Sohamdg081992) [reply otelcollector/fluent-bit/fluent-bit-daemonset.conf:168]: Thanks Rashmi! yes I will add this.

**Comment by Sohamdg081992** [comment]
> promColCPUUsage



> @Sohamdg081992 - Could you also pls check on the promColCPUUsage metric value for memory. It seems to be reporting incorrect values. I can create a bug separate from this PR too, since it is not related to this PR.

Thanks Rashmi! I have created a [bug](https://msazure.visualstudio.com/InfrastructureInsights/_sprints/taskboard/Prometheus-agent/InfrastructureInsights/Dilithium/CY24Q2/Monthly/04%20Apr%20(Apr%2001%20-%20Apr%2030)?workitem=27487819) in current sprint to track this separately.

File: otelcollector/fluent-bit/fluent-bit-daemonset.conf (hunk 4 above)
Comment (Sohamdg081992) [reply otelcollector/fluent-bit/fluent-bit-daemonset.conf:175]: I have added this change and have uploaded the screenshot of flowing telemetry(last screenshot).

**Comment by rashmichandrashekar** [comment]
Thanks @Sohamdg081992. Could you also make sure the metric - opentelemetry_allocator_collectors_discovered is flowing with all the custom dimensions including job name are flowing. or a screenshot with customdimensions in tgf and fbit would be good?
Also, what are the metrics? Are these for the vms? - 
otelcolVMRSSAvg 
otelcolVMRSS95 
meVMRSSAvg
meVMRSS95

**Comment by Sohamdg081992** [comment]
> otelcolVMRSSAvg

Thanks Rashmi! Those metrics are MemoryRSS values for otelcollector and ME processes. I have added the screenshots for the operator (Telegraf vs. FB)

**Comment by Sohamdg081992** [comment]
Updated the metric name based screenshots of Telegraf and Fluent Bit , showing both are same.

**Review by vishiy — APPROVED** [review]

**Pull request merged**